packages/useScript/src/useScript.ts

```diff
@@ -1,18 +1,53 @@
-import { useEffect, useState } from 'react'
+import { useEffect, useReducer } from 'react'
+
+type State = {
+  ready: boolean
+  error?: boolean
+}
+
+type Action = {
+  type: string
+}
+
+const SCRIPT_LOADED: string = 'SCRIPT_LOADED'
```

Comment:
I would prefer converting these `type` to an Enum, and point to those as the `type` inside the `Action`.

```
enum ScriptStatus = {
 LOADED = 'loaded',
 ERROR = 'error',
}

type Action = {
  type: ScriptStatus
}
```

Reply:
Makes sense! Not really into `TypeScript`, so I am open to improvements. ;)

Check my last commits. :)